src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -1185,6 +1197,64 @@ class query extends viewModelBase {
         });
     }
     
+    private onSpatialLoaded(queryResults: pagedResultExtended<document>) {
+        this.isSpatialQuery(false);
+        this.totalNumberOfMarkers(0);
+        
+        const spatialProperties = queryResults.additionalResultInfo.SpatialProperties;
+        if (spatialProperties && queryResults.items.length) {
+            this.isSpatialQuery(true);
+
+            // Each spatial markers model will contain the layer of markers per spatial properties pair
+            const spatialMarkersLayers: spatialMarkersLayerModel[] = [];
+            const spatialCirclesLayer: spatialCircleModel[] = [];
+            const spatialPolygonsLayer: spatialPolygonModel[] = [];
+
+            for (let i = 0; i < spatialProperties.length; i++) {
+                const latitudeProperty = spatialProperties[i].LatitudeProperty;
+                const longitudeProperty = spatialProperties[i].LongitudeProperty;
+
+                let pointsArray: geoPointInfo[] = [];
+                for (let i = 0; i < queryResults.items.length; i++) {
+                    const item = queryResults.items[i];
+                    const flatItem = generalUtils.flattenObj(item, "");
+                    
+                    const latitudeValue = _.get(flatItem, latitudeProperty) as number;
+                    const longitudeValue = _.get(flatItem, longitudeProperty) as number;
+                    
+                    if (latitudeValue != null && longitudeValue != null) {
+                        const point: geoPointInfo = { Latitude: latitudeValue, Longitude: longitudeValue, PopupContent: item };
+                        pointsArray.push(point);
+                        this.totalNumberOfMarkers(this.totalNumberOfMarkers() + 1);
```

Comment:
please don't update observable in for loop - just count this using variable and update one time.

Reply:
done